library/src/app/components/trade/trade.component.html

```diff
@@ -0,0 +1,102 @@
+<div
+  [ngClass]="{
+    'cybrid-dark-theme': (configService.config$ | async)?.theme === 'DARK'
+  }"
+>
+  <ng-container *ngIf="(isRecoverable$ | async) !== false; else error">
+    <ng-container *ngIf="(isLoading$ | async) !== true">
+      <ng-container [formGroup]="quoteGroup" *ngIf="quote$ | async as quote">
+        <div fxLayout="column" fxLayoutAlign="center">
+          <mat-form-field appearance="outline">
+            <mat-label>
+              <span>{{ 'trade.amount' | translate }}</span>
+              <span class="code">
+                {{
+                  input === 'fiat' ? quote.asset.code : quote.counterAsset.code
+                }}
+              </span>
+            </mat-label>
+            <input
+              matInput
+              #amount="matInput"
+              formControlName="amount"
+              type="number"
+              placeholder="0"
```

Comment:
Can negative numbers be input?

Reply:
Good catch. I've restricted it to positive numbers only. The 'buy' button is now disabled unless the amount is valid and > 0. I've also added an error message to the input field for this case.